src/hooks/fire/useLiveCollection.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useMemo, useState } from "react";
+import { QueryConstraint } from "@firebase/firestore";
+import {
+  collection,
+  FirestoreError,
+  getFirestore,
+  onSnapshot,
+  query,
+  QueryDocumentSnapshot,
+  QuerySnapshot,
+} from "firebase/firestore";
+import { isEqual } from "lodash";
+
+import { ALWAYS_EMPTY_ARRAY } from "settings";
+
+import { withIdConverter } from "utils/converters";
+import { withId } from "utils/id";
+
+type UseLiveCollectionOptions =
+  | string[]
+  | {
+      path: string[];
+      constraints?: (QueryConstraint | null | undefined)[];
+    };
+
+const toDataWithId = <T extends object>(d: QueryDocumentSnapshot<T>) =>
+  withId(d.data(), d.id);
+
+const isValidSegment = (segment: string | undefined): segment is string =>
+  "string" === typeof segment && "" !== segment;
+
+const isValidConstraint = (
+  constraint: QueryConstraint | null | undefined
+): constraint is QueryConstraint =>
+  constraint !== null && constraint !== undefined;
+
+export const useLiveCollection = <T extends object>(
```

Comment:
I think this could do with a comment saying why you'd use this one over all the other alternatives that we have in the codebase

Reply:
Similarly for the other new hooks.

Reply:
The deprecated ones have a comment.

Reply:
Perfect